### PR TITLE
Add Windows support

### LINKED ISSUE #23: Add windows CI action
Some tests and incompatibilities are found when running KPOps on a Windows machine. It would be nice to solve them and use the CI to keep KPOps working on windows.

### LINKED ISSUE #24: `signal.SIGALRM` not working in Python on Windows
We are using the signal module for the Kafka Connect timeouts. According to this [link](https://stackoverflow.com/questions/52779920/why-is-signal-sigalrm-not-working-in-python-on-windows), Windows does not implement this signal and we get:

```console
AttributeError: module 'signal' has no attribute 'SIGALRM'. Did you mean: 'SIGABRT'
```

### LINKED ISSUE #102: Variable substitution does not work on Windows
We use os.environ in order to substitute internal variables such as `pipeline_name`: https://github.com/bakdata/kpops/blob/8f5e1bdc7f4ddf13e633259908fd1d9bcc4b34e0/kpops/pipeline_generator/pipeline.py#L90. However, when calling `dict(os.environ)`, Windows returns a dictionary containing `{"PIPELINE_NAME": ...}`. Therefore, substitution does not work. We should find another way to do the substitution. In my opinion, we should not add our internal variables to os.environ but rather add os.environ to our dictionary of variables.

### PR DESCRIPTION
**Pull request by irux**
Closes #23 
Closes #24
Closes #102

### CODE REVIEW

**Comment by disrupted** [comment]
please reference the associated issues

File: kpops/utils/environment.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import os
+import platform
+from collections import UserDict
+from typing import Callable
+
+
+class Environment(UserDict):
+    def __init__(self, mapping=None, /, **kwargs) -> None:
+
```
Comment (disrupted) [inline kpops/utils/environment.py:9]: delete empty line

File: kpops/utils/environment.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    @staticmethod
+    def __key_camel_case_transform(key: str) -> str:
+        return key.lower()
+
+    @staticmethod
+    def __key_identity_transform(key: str) -> str:
+        return key
+
+    def __get_transformation(self) -> Callable[[str], str]:
```
Comment (disrupted) [inline kpops/utils/environment.py:29]: this looks like a staticmethod

**Review by disrupted — CHANGES_REQUESTED** [review]

**Review by raminqaf — DISMISSED** [review]
LGTM!

**Review by sujuka99 — DISMISSED** [review]
LGTM

**Review by disrupted — DISMISSED** [review]

**Review by sujuka99 — APPROVED** [review]

**Pull request merged**